src/storage/copier/csv_reader.cpp

```diff
@@ -0,0 +1,515 @@
+        columnSizes[column]++;
+    } break;
+    default:
+        throw CopyException("Unsupported data type " +
+                            LogicalTypeUtils::dataTypeToString(type.getLogicalTypeID()) +
+                            " inside LIST");
+    }
+    parse_chunk.getValueVector(column)->state->selVector->selectedSize++;
+    // move to the next column
+    column++;
+}
+
+bool BaseCSVReader::AddRow(
+    DataChunk& insert_chunk, column_id_t& column, std::string& error_message, uint64_t buffer_idx) {
+    if (mode == ParserMode::PARSING_HEADER) {
+        return true;
+    }
+    linenr++;
+    if (row_empty) {
+        row_empty = false;
+        if (return_types.size() != 1) {
+            if (mode == ParserMode::PARSING) {
+                // Set This position to be null
+                ;
+            }
+            column = 0;
+            return false;
+        }
+    }
+    if (column < return_types.size()) {
+        throw CopyException(StringUtils::string_format(
+            "Error in file {} on line {}: expected {} values per row, but got {}", filePath, linenr,
+            return_types.size(), column));
+    }
+    if (mode == ParserMode::PARSING &&
+        columnSizes[0] >= common::StorageConstants::NODE_GROUP_SIZE) {
+        Flush(insert_chunk, buffer_idx);
+        return true;
+    }
+
+    column = 0;
+    return false;
+}
+
+bool BaseCSVReader::Flush(DataChunk& insert_chunk, uint64_t buffer_idx, bool try_add_line) {
+    if (parse_chunk.getNumValueVectors() == 0) {
+        return true;
+    }
+    for (uint64_t c = 0; c < parse_chunk.getNumValueVectors(); c++) {
+        auto parse_vector = parse_chunk.getValueVector(c);
+        auto result_vector = insert_chunk.getValueVector(c);
+        insert_chunk.getValueVector(c)->state->selVector->selectedSize =
+            parse_chunk.getValueVector(c)->state->selVector->selectedSize;
+        insert_chunk.insert(c, parse_chunk.getValueVector(c));
+    }
+    InitParseChunk(parse_chunk.getNumValueVectors());
+    return true;
+}
+
+void BaseCSVReader::InitParseChunk(column_id_t numCols) {
+    parse_chunk = DataChunk(numCols);
+}
+
+BufferedCSVReader::BufferedCSVReader(const std::string& filePath,
+    common::CSVReaderConfig* csvReaderConfig, catalog::TableSchema* tableSchema,
+    storage::MemoryManager* memoryManager = nullptr)
+    : BaseCSVReader(filePath, csvReaderConfig, tableSchema), bufferSize(0), position(0), start(0) {
+    Initialize(tableSchema->getProperties(), memoryManager);
+}
+
+void BufferedCSVReader::Initialize(
+    std::vector<kuzu::catalog::Property*> properties, storage::MemoryManager* memoryManager) {
+    // PrepareComplexParser();
+    for (auto property : properties) {
+        return_types.push_back(*property->getDataType());
+    }
+    fd = open(filePath.c_str(), O_RDONLY);
+    ResetBuffer();
+    ReadHeader(csvReaderConfig->hasHeader);
+    // Initializing parse_chunk DataChunk
+    parse_chunk = DataChunk(properties.size());
```

Comment:
Again, why are we initializing the parse_chunk in multiple places? It looks like you are initializing parse_chunk correctly here. So make sure this is the right place, and clean up other places.